lib/wrapping_integer.fz

```diff
@@ -194,6 +194,20 @@ wrapping_integer(redef W type : wrapping_integer W) : integer W, wrapping_intege
     thiz ^ all_bits_set
 
 
+  # count the number of 1 bits in the binary representation of this
+  # integer.
+  #
+  ones_count i32 is abstract
+
+
+  # this integer as an array of bytes
+  as_bytes array u8 is
+    array u8 bytes (idx ->
+      shift W := from_u32((bytes.as_u32-idx.as_u32-1)*8)
+      (wrapping_integer.this>>shift).low8bits)
```

Comment:
This needs to document what byte ordering is used (little or big endian or platform dependent?). My preference is little endian as default, since then you can read it as a different size integer and you get the same value (provided it fits or padded with zeros). 

For integers in .fum-files, we use little endian for Fuzion constants and big endian for other integers (I don't know why I did that, maybe Java influence), this should all be little endian except for the magic numbers to identify files (MIR_FILE_MAGIC etc.).